src/TLuaInterpreter.cpp

```diff
@@ -7048,6 +7088,129 @@ int TLuaInterpreter::permSubstringTrigger( lua_State * L )
     return 1;
 }
 
+int TLuaInterpreter::permKey( lua_State *L )
+{
+    string luaName;
+    if( ! lua_isstring( L, 1 ) )
+    {
+        lua_pushstring( L, "permKey: need a name for this key" );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaName = lua_tostring( L, 1 );
+    }
+
+    string luaParent;
+    if( ! lua_isstring( L, 2 ) )
+    {
+        lua_pushstring( L, "permKey: need a parent key/group to add this key to" );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaParent = lua_tostring( L, 2 );
+    }
+
+    int luaKeyCode;
+    if( ! lua_isstring( L, 3 ) )
+    {
+        lua_pushstring( L, "permKey: need the keycode for the key" );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaKeyCode = lua_tointeger( L, 3 );
+    }
+
+    int luaModifier;
```

Comment:
We may need to document the integer values for this - they are Qt defined constants:
>`enum Qt::KeyboardModifier`
`flags Qt::KeyboardModifiers`
>This enum describes the modifier keys.
>
    Constant                Value         Description
    Qt::NoModifier          0x00000000    No modifier key is pressed.
    Qt::ShiftModifier       0x02000000    A Shift key on the keyboard is pressed.
    Qt::ControlModifier     0x04000000    A Ctrl key on the keyboard is pressed.
    Qt::AltModifier         0x08000000    An Alt key on the keyboard is pressed.
    Qt::MetaModifier        0x10000000    A Meta key on the keyboard is pressed.
    Qt::KeypadModifier      0x20000000    A keypad button is pressed.
    Qt::GroupSwitchModifier 0x40000000    X11 only. A Mode_switch key on the keyboard is pressed.

Note that on MacOs that Ctrl is the "Command" or "⌘" key...